Sources/Subs-Calendar.php

```diff
@@ -1851,4 +1851,25 @@ function removeHolidays($holiday_ids)
 	));
 }
 
+/**
+ * Helper function to convert date string to english
+ * so that date_parse can parse the date
+ *
+ * @param string $date A localized date string
+ * @return string English date string
+ */
+function convertDateToEnglish($date)
+{
+	global $txt, $context;
+
+	if ($context['user']['language'] == 'english')
+		return $date;
+
+	return strtr($date, array_combine(array_merge($txt['months_titles'], $txt['months_short'], $txt['days'], $txt['days_short']),
+		array_merge(array('January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October',	'November', 'December'),
+			array('Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'),
+			array('Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'),
+			array('Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'))));
```

Comment:
```suggestion
	$replacements = array_combine($txt['months_titles'], array(
		'January', 'February', 'March', 'April', 'May', 'June',
		'July', 'August', 'September', 'October',	'November', 'December'
	));
	$replacements += array_combine($txt['months_short'], array(
		'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
		'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
	));
	$replacements += array_combine($txt['days'], array(
		'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'
	));
	$replacements += array_combine($txt['days_short'], array(
		'Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'
	));

	$replacements[$txt['time_am']] = 'AM';
	$replacements[$txt['time_pm']] = 'PM';
	if (strftime('%p', strtotime('01:00:00')) !== 'p')
	{
		$replacements[strftime('%p', strtotime('01:00:00'))] = 'AM';
		$replacements[strftime('%p', strtotime('23:00:00'))] = 'PM';
	}
	if (strftime('%P', strtotime('01:00:00')) !== 'P')
	{
		$replacements[strftime('%P', strtotime('01:00:00'))] = 'AM';
		$replacements[strftime('%P', strtotime('23:00:00'))] = 'PM';
	}

	return strtr($date, $replacements);
```

Reply:
Why do you need this part strftime('%p', strtotime('01:00:00')?
To get the system am/pm string? Is that string every used?

!== 'p' Will it return 'p' if it is not supported? or '%p'?

Reply:
> Why do you need this part strftime('%p', strtotime('01:00:00')?
> To get the system am/pm string? Is that string every used?

Yes, that is why. If you look at `timeformat()` in Subs.php, you will see that we only use `$txt['time_am']` and `$txt['time_pm']` as fallbacks in the event that `%p` is not supported in the current locale. If `%p` is supported, then we use the output of `strftime('%p')`.

> !== 'p' Will it return 'p' if it is not supported? or '%p'?

That is correct. It will return `p` if `%p` is not supported.

... Except on Windows, where `strftime()` will return false if given an unsupported format specifier. I forgot about that in my suggested code above. I'll add a new suggestion to deal with that in a review of your latest commit.